java/vector/src/main/java/org/apache/arrow/vector/complex/NonNullableStructVector.java

```diff
@@ -99,6 +99,17 @@ public void setInitialCapacity(int numRecords) {
     }
   }
 
+  @Override
+  public void setInitialCapacity(int valueCount, double density) {
+    for (final ValueVector vector : (Iterable<ValueVector>) this) {
+      if (vector instanceof DensityAwareVector) {
```

Comment:
What do you think of having all vectors implement
`setInitialCapacity(valueCount, density)` and delegate to `setInitialCapacity(valueCount)`
instead of case matching here?

Reply:
I think that's unnecessary delegation and then there is no purpose of having a DensityAwareVector interface since typically everyone will implement the interface -- some will delegate and rest will implement.

Reply:
I agree, I think it's best to just check the class instance where needed instead of delegating

Reply:
Ok, SGTM.